### PR TITLE
Add new service auth token as a header to outbound http requests

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

See title

### How to review

Check changes make sense. Until all services have been updated with new authentication, it will be hard to test successfully.

### Who can review

Team B

### CODE REVIEW

File: config/importer.go [hunk 1]
```diff
@@ -12,36 +12,31 @@ import (
 // Config struct to hold application configuration.
 type Config struct {
 	BindAddr                       string        `envconfig:"BIND_ADDR"`
+	ServiceAuthToken               string        `envconfig:"SERVICE_AUTH_TOKEN"                     json:"-"`
```
Comment (Matt-Guest) [inline config/importer.go:15]: Can the README be updated as well

File: config/importer.go [hunk 2]
```diff
@@ -12,36 +12,31 @@ import (
...
 	KafkaAddr                      []string      `envconfig:"KAFKA_ADDR"`
 	IncomingInstancesTopic         string        `envconfig:"DIMENSIONS_EXTRACTED_TOPIC"`
 	IncomingInstancesConsumerGroup string        `envconfig:"DIMENSIONS_EXTRACTED_CONSUMER_GROUP"`
 	OutgoingInstancesTopic         string        `envconfig:"DIMENSIONS_INSERTED_TOPIC"`
 	EventReporterTopic             string        `envconfig:"EVENT_REPORTER_TOPIC"`
 	DatasetAPIAddr                 string        `envconfig:"DATASET_API_ADDR"`
-	DatasetAPIAuthToken            string        `envconfig:"DATASET_API_AUTH_TOKEN"`
+	DatasetAPIAuthToken            string        `envconfig:"DATASET_API_AUTH_TOKEN"                 json:"-"`
 	DatabaseURL                    string        `envconfig:"DB_URL"`
```
Comment (Matt-Guest) [inline config/importer.go]: The database url shouldn't be logged as well, as it contains the user and password to neo4j

**Review by Matt-Guest — COMMENTED** [review]
On startup the app should call the /identity with just the service token. So that we know its a valid token, if its not we should log an error and close the app.

**Review by mattrout92 — APPROVED** [review]
LGTM

**Pull request merged**